loader_hub/file/unstructured/base.py

```diff
@@ -0,0 +1,35 @@
+"""Unstructured file reader.
+
+A parser for unstructured text files using Unstructured.io.
+Supports .txt, .docx, .pptx, .jpg, .png, .eml, .html, and .pdf documents.
+
+"""
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+
+class UnstructuredReader(BaseReader):
+    """General unstructured text reader for a variety of files."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Init params."""
+        super().__init__(*args, **kwargs)
+
+        # Prerequisite for Unstructured.io to work
+        import nltk
+
+        nltk.download("averaged_perceptron_tagger")
+
+    def load_data(
+        self, file: Path, extra_info: Optional[Dict] = None
+    ) -> List[Document]:
+        """Parse file."""
+        from unstructured.partition.auto import partition
```

Comment:
we could probably start not doing lazy imports since we have a requirements.txt file now

Reply:
IMO still better to do lazy. Otherwise, for example, loading SimpleDirectoryReader would import a million things. Having the requirements txt at least eliminates the need to do try/except to check if things are installed. What do you think?

Reply:
Actually I refactored SimpleDirectoryReader so that it doesn't do that anymore. But still feels safer to not be importing unnecessarily.

Reply:
oh ok. that's fair, yeah not a big deal for now